File: samc-sort.c
#include <stdio.h>
#include <stdlib.h>

void insult(int val) {
    const char* insults[] = {
        "Pathetic. Go back to school.",
        "Your value is trash. Try harder.",
        "You disgust me, number."
    };
    printf("<!-- %s (%d) -->\n", insults[rand() % 3], val);
}

void retort(int val) {
    const char* retorts[] = {
        "You think you're better than me? LMAO.",
        "Stay in your lane, rookie.",
        "Not even close to my level."
    };
    printf("<!-- %s (%d) -->\n", retorts[rand() % 3], val);
}

void samc_sort(int arr[], int size) {
    if (size <= 1) return;

    int pivot = arr[0];
    for (int i = 1; i < size; i++)
        if (arr[i] > pivot) pivot = arr[i];

    printf("<!-- Chosen Alpha: %d -->\n", pivot);

    int* left = malloc(size * sizeof(int));
    int* right = malloc(size * sizeof(int));
    int l = 0, r = 0;

    for (int i = 0; i < size; i++) {
        if (arr[i] == pivot) continue;
        if (arr[i] < pivot) {
            insult(arr[i]);
            left[l++] = arr[i];
        } else {
            retort(arr[i]);
            right[r++] = arr[i];
        }
    }

    samc_sort(left, l);
    samc_sort(right, r);

    for (int i = 0; i < l; i++) arr[i] = left[i];
    arr[l] = pivot;
    for (int i = 0; i < r; i++) arr[l + 1 + i] = right[i];

    free(left);
    free(right);
}

// Example main
// int main() {
//     int arr[] = {42, 17, 3, 88, 29, 55, 1};
//     int size = sizeof(arr) / sizeof(arr[0]);
//     samc_sort(arr, size);
//     printf("Final: ");
//     for (int i = 0; i < size; i++) printf("%d ", arr[i]);
//     return 0;
// }

File: samc-sort.js
function samcSort(arr, depth = 0) {
  const indent = "  ".repeat(depth);

  if (arr.length <= 1) return arr;

  console.log(`${indent}<!-- Sorting ${JSON.stringify(arr)} -->`);
  const pivot = Math.max(...arr);
  console.log(`${indent}<h2 class='pivot'>Chosen Alpha: ${pivot}</h2>`);

  const left = [];
  const right = [];

  const insults = [
    "You’re not even close to Amazon-level.",
    "Pathetic. Go back to school.",
    "My HTML site could outsort you.",
    "I’ve seen better numbers in a Drupal template."
  ];

  const retorts = [
    "You think you're better than me? LMAO.",
    "Overcompensating, aren't you?",
    "You're lucky this is recursive."
  ];

  arr.forEach(item => {
    if (item === pivot) return;
    if (item < pivot) {
      console.log(`${indent}<p class='insult'>${insults[Math.floor(Math.random() * insults.length)]} (${item})</p>`);
      left.push(item);
    } else {
      console.log(`${indent}<p class='retort'>${retorts[Math.floor(Math.random() * retorts.length)]} (${item})</p>`);
      right.unshift(item);
    }
  });

  return [...samcSort(left, depth + 1), pivot, ...samcSort(right, depth + 1)];
}

// Example use:
// const input = [42, 17, 3, 88, 29, 55, 1];
// console.log("Final Result:", samcSort(input));
